### DISCUSSION TITLE
New Block / Extension - Save screen shot - Generative Art

### DISCUSSION THREAD

**Discussion by gpg342** [ScratchAddons/ScratchAddons#4379] [Ideas]
Context -- K12 computer science teacher using Scratch for the purpose of generative art / creativity.  For example, here I use Scratch to draw a 5x5 grid and stamp the prime numbers differently -- [https://scratch.mit.edu/projects/656108316](https://scratch.mit.edu/projects/656108316).

I would like to programmatically save an image of the canvas (say to PNG)   -- that way I can get the students to generate a range of images, and save them to a folder for later processing.  I know this can be done manually.

Does such a block / extension exist?

Cheers
Glen

**Comment by cobaltt7** [comment]
We will not be adding new blocks as they will not work for non-SA users


________________________________
From: gpg342 ***@***.***>
Sent: Tuesday, March 8, 2022 6:07:06 AM
To: ScratchAddons/ScratchAddons ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ScratchAddons/ScratchAddons] New Block / Extension - Save screen shot - Generative Art (Discussion #4379)


Context -- K12 computer science teacher using Scratch for the purpose of generative art / creativity. For example, here I use Scratch to draw a 5x5 grid and stamp the prime numbers differently -- https://scratch.mit.edu/projects/656108316.

I would like to programmatically save an image of the canvas (say to PNG) -- that way I can get the students to generate a range of images, and save them to a folder for later processing. I know this can be done manually.

Does such a block / extension exist?

Cheers
Glen

**Reply by gpg342** [reply]
Ok - appreciate that.

I'll keep Googling.
G
